dev/com.ibm.ws.session.cache/src/com/ibm/ws/session/store/cache/CacheHashMap.java

```diff
@@ -133,7 +136,7 @@ public class CacheHashMap extends BackedHashMap {
         appDataTablesPerThread = (!_smc.writeAllProperties() && !_smc.getEnableTimeBasedWrite());
 
         AccessController.doPrivileged((PrivilegedAction<Void>) () -> {
-            cacheInit();
+            CheckpointPhase.onRestore(() -> cacheInit());
```

Comment:
For InstantOn we do not support Java 2 security.  But this code pattern will confuse others because for the normal (non-checkpoint) case the call to `cacheInit` will be done directly in the invocation of `doPrivileged`.  But in the checkpoint case the call to `cacheInit` will happen at restore outside of the context of this `doPrivileged` call.  I think this should be changed to:

```java
        CheckpointPhase.onRestore(() -> AccessController.doPrivileged((PrivilegedAction<Void>) () -> {
            cacheInit();
            return null;
        }));
```

Not sure that will compile, but the basic change is that we always surround the call to `cacheInit()` with a `doPrivileged`